### PR TITLE
Topic/parallel

### PR DESCRIPTION
**Pull request by danielocfb**

### CODE REVIEW

File: ci/vmtest/run_selftests.sh [hunk 1]
```diff
@@ -54,21 +48,34 @@ read_test_names() {
...
 }
 
-test_progs() {
-  foldable start test_progs "Testing test_progs"
+test_progs_helper() {
+  local suffix="$1"
+  local args="$2"
+  local selftest="test_progs${suffix}"
+
+  foldable start test_progs${suffix} "Testing test_progs${suffix}"
```
Comment (mykola-lysenko) [inline ci/vmtest/run_selftests.sh:56]: minor: can we reuse $selftest variable here?

File: ci/vmtest/run_selftests.sh [hunk 2]
```diff
@@ -54,21 +48,34 @@ read_test_names() {
     foldable end read_test_names
 }
 
-test_progs() {
-  foldable start test_progs "Testing test_progs"
+test_progs_helper() {
+  local suffix="$1"
```
Comment (mykola-lysenko) [inline ci/vmtest/run_selftests.sh:52]: minor: if we address below comment, do we need suffix variable or can move it there?

**Comment by mykola-lysenko** [comment]
thanks Daniel, other then two minor comments, everything looks good. waiting until tomorrow to see tests pass and confirm that failed parallel tests not to fail the overall test result

File: ci/vmtest/run_selftests.sh (hunk 1 above)
Comment (danielocfb) [reply ci/vmtest/run_selftests.sh:56]: Indeed, we should.

File: ci/vmtest/run_selftests.sh (hunk 2 above)
Comment (danielocfb) [reply ci/vmtest/run_selftests.sh:52]: Agreed.

**Review by mykola-lysenko — APPROVED** [review]

**Comment by danielocfb** [comment]
So the problem I see is parallel runs seem to trigger hangs/deadlocks/livelocks in the kernel causing us to time out. And those timeouts actually manifest as an error after all.

**Comment by danielocfb** [comment]
> So the problem I see is parallel runs seem to trigger hangs/deadlocks/livelocks in the kernel causing us to time out. And those timeouts actually manifest as an error after all.

In addition to the change that enabled panics on soft & hard lockups, I've added a patch that enables hung task detection. From my perspective, that resolves these issues.

@mykola-lysenko can I ask for another review, please?

**Comment by mykola-lysenko** [comment]
> > So the problem I see is parallel runs seem to trigger hangs/deadlocks/livelocks in the kernel causing us to time out. And those timeouts actually manifest as an error after all.
> 
> In addition to the change that enabled panics on soft & hard lockups, I've added a patch that enables hung task detection. From my perspective, that resolves these issues.
> 
> @mykola-lysenko can I ask for another review, please?

so, basically, because we are not going to timeout jobs anymore, they will fail and this will make the run to pass, correct?

**Comment by mykola-lysenko** [comment]
> > > So the problem I see is parallel runs seem to trigger hangs/deadlocks/livelocks in the kernel causing us to time out. And those timeouts actually manifest as an error after all.
> > 
> > 
> > In addition to the change that enabled panics on soft & hard lockups, I've added a patch that enables hung task detection. From my perspective, that resolves these issues.
> > @mykola-lysenko can I ask for another review, please?
> 
> so, basically, because we are not going to timeout jobs anymore, they will fail and this will make the run to pass, correct?

if yes, this sounds good to me

I also see that Alexei accepted your patch to bpf-next, but I guess it is here to cover bpf as well, right?

**Review by mykola-lysenko — APPROVED** [review]
Change looks good to me!

Here, we added two diffs as well. One is already upstreamed in bpf-next and will be removed once it is in bpf as well. Another one needs more testing in the CI before upstreaming.

Before merging, let's ensure test run passes and I would love to get my questions answered.

Thanks @d-e-s-o !

**Comment by danielocfb** [comment]
> > ...
> so, basically, because we are not going to timeout jobs anymore, they will fail and this will make the run to pass, correct?

Yes, exactly.

> I also see that Alexei accepted your patch to bpf-next, but I guess it is here to cover bpf as well, right?

That is correct as well. Otherwise the `0002-*` patch would not apply cleanly on `bpf` (and we would not have the soft & hard lockup detection, of course)

**Pull request merged**